llvm/include/llvm/Frontend/OpenMP/OMP.td

```diff
@@ -142,6 +142,20 @@ def OMPC_Schedule : Clause<"schedule"> {
   ];
 }
 
+def OMP_SCHEDULE_MOD_None : ClauseVal<"None", 0, 1> { let isDefault = 1; }
+def OMP_SCHEDULE_MOD_Monotonic : ClauseVal<"Monotonic", 1, 1> {}
+def OMP_SCHEDULE_MOD_Nonmonotonic : ClauseVal<"Nonmonotonic", 2, 1> {}
+
+def OMPC_Schedule_modifier : Clause<"schedule_modifier"> {
+  let flangClass = "OmpScheduleModifier";
+  let enumClauseValue = "ScheduleModifier";
+  let allowedClauseValues = [
+    OMP_SCHEDULE_MOD_None,
+    OMP_SCHEDULE_MOD_Monotonic,
+    OMP_SCHEDULE_MOD_Nonmonotonic,
+  ];
+}
+
```

Comment:
modifier is not a clause, so we cannot directly re-use the clause mechanism. I think we can either directly write this as string enums in mlir OpenMPOps.td or create a new kind of representation in this td file.